### PR TITLE
Ssa Qa Fixes

### PR DESCRIPTION
**Pull request by joeizang**
address issues highlighted in QA by @Aniket-Engg

### CODE REVIEW

File: libs/remix-ui/static-analyser/src/lib/components/BasicTitle.tsx [hunk 1]
```diff
@@ -30,15 +31,21 @@ export function calculateWarningStateEntries(entries: [string, any][]) {
...
+  }, [props.warningStateEntries.length])
   return (
-    <span>Remix{props.warningStateEntries.length > 0 ?   !props.hideWarnings ? <i data-id="StaticAnalysisErrorCount" className={`badge ${calculateWarningStateEntries(props.warningStateEntries).length > 0 ? 'badge-warning' : 'badge-danger'} rounded-circle ml-1 text-center`}>{calculateWarningStateEntries(props.warningStateEntries).length}</i>: (
-      <i className="badge badge-warning rounded-circle ml-1 text-center">
-        {
-          calculateWarningStateEntries(props.warningStateEntries).errors.length
-        }
-      </i>
-    ) : null}
+    <span>Remix{props.warningStateEntries.length > 0 ?   !props.hideWarnings ? !props.showLibsWarnings ?  <i data-id="StaticAnalysisErrorCount" className={`badge ${calculateWarningStateEntries(props.warningStateEntries).length > 0 ? 'badge-warning' : 'badge-danger'} rounded-circle ml-1 text-center`}>
```
Comment (LianaHus) [inline libs/remix-ui/static-analyser/src/lib/components/BasicTitle.tsx:40]: extra spaces and the line is too long. could you plz fix this?

File: libs/remix-ui/static-analyser/src/lib/remix-ui-static-analyser.tsx [hunk 2]
```diff
@@ -760,14 +733,16 @@ export const RemixUiStaticAnalyser = (props: RemixUiStaticAnalyserProps) => {
...
+                solhintEnabled ? <Tab
+                  key={tabKeys[0].tabKey}
+                  title={tabKeys[0].title}
+                  eventKey={tabKeys[0].tabKey}
+                  tabClassName="text-decoration-none font-weight-bold px-2"
+                >
+                  {tabKeys[0].child}
+                </Tab> : null
+              }
               { showSlither && slitherEnabled ? <Tab
```
Comment (LianaHus) [inline libs/remix-ui/static-analyser/src/lib/remix-ui-static-analyser.tsx:746]: New line required to make it consistent with others.
Also here https://github.com/ethereum/remix-project/pull/3838/files#diff-a9943b741f29f3e349b333686b2558938e5ff1c853cd177eb6d8bd28e60be3a2R629

File: libs/remix-ui/vertical-icons-panel/src/lib/components/Badge.tsx [hunk 3]
```diff
@@ -58,7 +64,7 @@ function Badge ({ badgeStatus }: BadgeProps) {
                     <i className="far fa-exclamation-triangle"></i></span>
```
Comment (LianaHus) [inline libs/remix-ui/vertical-icons-panel/src/lib/components/Badge.tsx:64]: wrong indent for the scope

File: libs/remix-ui/vertical-icons-panel/src/lib/components/Badge.tsx [hunk 4]
```diff
@@ -58,7 +64,7 @@ function Badge ({ badgeStatus }: BadgeProps) {
...
                     : <span>&nbsp;</span> : badgeStatus.text }
             </i>
-        ) : null
+          </CustomTooltip>) : null
```
Comment (LianaHus) [inline libs/remix-ui/vertical-icons-panel/src/lib/components/Badge.tsx:67]: seems you have an extra space here

File: libs/remix-ui/static-analyser/src/lib/remix-ui-static-analyser.tsx (hunk 2 above)
Comment (joeizang) [reply libs/remix-ui/static-analyser/src/lib/remix-ui-static-analyser.tsx:753]: Done

**Review by Aniket-Engg — COMMENTED** [review]
1. SolHint should be used as Solhint as each place
2. Tooltip on Solhint says: `Solhint lints the selected contract`. It seems a bit odd as not contract selection dropdown is there. May be we can just use official definition: `Solhint lints Solidity code for security and style guide validations`
3. If I just compile a contract and does not run the analysis, badge on side panel is green.
4. In the result section, Solhint is still written as `Linter`
5. Error type that is error or warning is still shown on warning description when clicked on `Hide Warnings`
6. Even after compiling a contract, tooltip on analyse button says `To run analysis tools, first compile a contract`
Screenshot 2023-06-22 at 7 45 30 PM
7. If another contract is compiled, it still shows warning badge on plugin icon and tooltip refer to previous compilation warnings

File: libs/remix-ui/static-analyser/src/lib/components/BasicTabBody.tsx [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+  warningState: any
+  ssaWarnings: any[]
+  ssaWarningsNoLibs: any[]
+  startAnalysis: boolean
+  analysisModule: any
+}
+export function BasicTabBody({ state, hideWarnings, ssaWarnings, ssaWarningsNoLibs, showLibsWarning, warningState, startAnalysis, analysisModule}: BasicTabBodyProps) {
+const test = ssaWarningsNoLibs.filter(x => x.options.isLibrary === false)
+const test1 = ssaWarningsNoLibs.filter(x => x.options.isLibrary)
+console.log({test, test1 })
```
Comment (Aniket-Engg) [inline libs/remix-ui/static-analyser/src/lib/components/BasicTabBody.tsx:17]: remove log

**Comment by Aniket-Engg** [comment]
I see debugger is activated while using SSA plugin. Is that required?

File: libs/remix-ui/static-analyser/src/lib/components/BasicTabBody.tsx (hunk 5 above)
Comment (joeizang) [reply libs/remix-ui/static-analyser/src/lib/components/BasicTabBody.tsx:17]: done

Comment (joeizang) [reply libs/remix-ui/static-analyser/src/lib/components/BasicTabBody.tsx:17]: done

**Review by ryestew — APPROVED** [review]

**Review by Aniket-Engg — CHANGES_REQUESTED** [review]
I compiled a file and checked the box `Show warnings for external libraries` without running the analysis, badge is green

Screenshot 2023-06-23 at 7 20 40 PM

**Review by Aniket-Engg — CHANGES_REQUESTED** [review]
Screenshot 2023-06-23 at 7 24 28 PM

Count on badge is incorrect

**Comment by Aniket-Engg** [comment]
Screenshot 2023-06-23 at 7 26 07 PM

If Solhint is disabled, Result for remix is not visible

**Comment by Aniket-Engg** [comment]
Click `Hide Warnings` box. Type of warning is not there anymore for Solhint warnings but empty space is there.
Screenshot 2023-06-23 at 7 34 26 PM

**Comment by Aniket-Engg** [comment]
If we unselect Remix and Linter both, Analyze button is still enabled

**Comment by Aniket-Engg** [comment]
Screenshot 2023-06-23 at 7 39 53 PM

If Remix is not selected, Analysis doesn't run for `Solhint`

**Comment by joeizang** [comment]
> I compiled a file and checked the box `Show warnings for external libraries` without running the analysis, badge is green
> 
> Screenshot 2023-06-23 at 7 20 40 PM

This is now fixed.

**Comment by joeizang** [comment]
> Click `Hide Warnings` box. Type of warning is not there anymore for Solhint warnings but empty space is there. Screenshot 2023-06-23 at 7 34 26 PM

This is now fixed

**Comment by joeizang** [comment]
> If we unselect Remix and Linter both, Analyze button is still enabled

This has been fixed

**Comment by joeizang** [comment]
> Screenshot 2023-06-23 at 7 24 28 PM
> 
> Count on badge is incorrect

This is now fixed

**Comment by ryestew** [comment]
If only Solhint is checked, there are no results after clicking on the analyze button.

and some small stuff:
When Remix and Solhint are checked, before analysis has been run, the results show "Nothing to report"
image

But if you just check Solhint, it doesn't show anything - which makes more sense to me.

And also, if there is nothing to report,and if we are using that line - should that be printed out when you hide warnings and have no errors? (yes more conditionals) - maybe we don't need the line "nothing to report"

**Comment by Aniket-Engg** [comment]
1. Analyze button is enabled if none of Remix and Solhint are selected
2. If I uncheck Solhint, I see green badge on plugin icon.
3. If I select only Solhint and not Remix and run analysis, it doesn't show result details
Screenshot 2023-06-28 at 1 59 00 PM
4. If I compile other file and does not run analysis, result section shows : `Last result for: <newly compile file name>`
Screenshot 2023-06-28 at 2 06 15 PM
 The strange thing is that if you click on warning, it highlights the location in newly compiled file not for the file for which analysis was run
5. Solhint warning again started showing error types that are warning or error in warning details
6. If Solhint is selected in result and it is unchecked from analyse, it hides Solhint from Result but doesn't show Remix result automatically
Screenshot 2023-06-28 at 2 16 14 PM

**Review by Aniket-Engg — COMMENTED** [review]
1. Analyze button is enabled if none of Remix and Solhint are selected
2. If I uncheck Solhint, I see green badge on plugin icon.
3. If I select only Solhint and not Remix and run analysis, it doesn't show result details
Screenshot 2023-06-28 at 1 59 00 PM
4. If I compile other file and does not run analysis, result section shows : `Last result for: <newly compile file name>`
Screenshot 2023-06-28 at 2 06 15 PM
 The strange thing is that if you click on warning, it highlights the location in newly compiled file not for the file for which analysis was run
5. Solhint warning again started showing error types that are warning or error in warning details
6. If Solhint is selected in result and it is unchecked from analyse, it hides Solhint from Result but doesn't show Remix result automatically
Screenshot 2023-06-28 at 2 16 14 PM

**Comment by ryestew** [comment]
Aniket's point 6 - if you uncheck solhint at the top of the page, it hides the label, but the warnings that were there are now part of the remix tab.  But then who is going to uncheck from the top of the page - they should be using the tabs.  So I vote that this is "over bullit-proofing" this and we should merge.  But it could be fixed - but it is not necessary.

File: apps/remix-ide/src/app/tabs/analysis-tab.js [hunk 6]
```diff
@@ -47,8 +57,18 @@ class AnalysisTab extends ViewPlugin {
     }
 
     this.event.register('staticAnaysisWarning', (count) => {
+      let payloadType = ''
+
+      this.hints && this.hints.length > 0 ? this.hints.forEach(hint => {
```
Comment (yann300) [inline apps/remix-ide/src/app/tabs/analysis-tab.js:62]: you could do a normal for/loop and break it as soon as you encounter an  error type, then you don't need to loop over the entire set.

**Comment by Aniket-Engg** [comment]
> Aniket's point 6 - if you uncheck solhint at the top of the page, it hides the label, but the warnings that were there are now part of the remix tab. But then who is going to uncheck from the top of the page - they should be using the tabs. So I vote that this is "over bullit-proofing" this and we should merge. But it could be fixed - but it is not necessary.

One of the tool can be unchecked to run the analysis for next contract and It will look really odd.

File: libs/remix-ui/static-analyser/src/lib/remix-ui-static-analyser.tsx [hunk 7]
```diff
@@ -81,6 +84,8 @@ export const RemixUiStaticAnalyser = (props: RemixUiStaticAnalyserProps) => {
   const [hints, setHints] = useState<SolHintReport[]>([])
   const [slitherWarnings, setSlitherWarnings] = useState([])
   const [ssaWarnings, setSsaWarnings] = useState([])
+  const [selectedTab, setSelectedTab] = useState<tabSelectionType>('remix')
+
```
Comment (LianaHus) [inline libs/remix-ui/static-analyser/src/lib/remix-ui-static-analyser.tsx:88]: extra line

File: libs/remix-ui/static-analyser/src/lib/remix-ui-static-analyser.tsx [hunk 8]
```diff
@@ -174,10 +187,18 @@ export const RemixUiStaticAnalyser = (props: RemixUiStaticAnalyserProps) => {
...
     return () => { }
   }, [props])
 
+  const hintErrors = hints.filter(hint => hint.type === 'error')
+  const noLibSlitherWarnings = slitherWarnings.filter(w => !w.options.isLibrary)
+  const slitherErrors = noLibSlitherWarnings.filter(slitherError => slitherError.options.type === 'error')
+  const remixAnalysisNoLibs = ssaWarnings.filter(ssa => ssa.options.isLibrary === false)
+  const remixAnalysisLessWarnings = ssaWarnings.filter(ssa => ssa.options.type !== 'warning')
+
+//#region PreviousCode (message, filterWarnings, showWarnings, handleCheckSingle, categoryItem, categorySection)
```
Comment (LianaHus) [inline libs/remix-ui/static-analyser/src/lib/remix-ui-static-analyser.tsx:201]: remove the comment plz

File: libs/remix-ui/static-analyser/src/lib/remix-ui-static-analyser.tsx [hunk 9]
```diff
@@ -372,18 +350,106 @@ export const RemixUiStaticAnalyser = (props: RemixUiStaticAnalyserProps) => {
       </div>
     )
   }
+//#endregion
```
Comment (LianaHus) [inline libs/remix-ui/static-analyser/src/lib/remix-ui-static-analyser.tsx:353]: rm ?

File: libs/remix-ui/static-analyser/src/lib/remix-ui-static-analyser.tsx [hunk 10]
```diff
@@ -372,18 +350,106 @@ export const RemixUiStaticAnalyser = (props: RemixUiStaticAnalyserProps) => {
...
+
+useEffect(() => {
+  if(!hideWarnings && !showLibsWarning) {
+    const slitherTotal = (slitherWarnings && state.data && state.source !== null  ? slitherWarnings.filter(x => !x.options.isLibrary && x.hasWarning).length : 0)
+    const ssaTotal = ssaWarnings && state.data && state.source !== null ? ssaWarnings.filter(x => !x.options.isLibrary && x.hasWarning).length : 0
+    const hintsTotal = hints && state.data && state.source !== null ? hints.length : 0
+    props.event.trigger('staticAnaysisWarning', [(slitherTotal + ssaTotal + hintsTotal) === 0 ? -1 : (slitherTotal + ssaTotal + hintsTotal)])
+  }
+  if(!hideWarnings && showLibsWarning) {
```
Comment (LianaHus) [inline libs/remix-ui/static-analyser/src/lib/remix-ui-static-analyser.tsx:362]: space after if

File: libs/remix-ui/static-analyser/src/lib/remix-ui-static-analyser.tsx [hunk 11]
```diff
@@ -372,18 +350,106 @@ export const RemixUiStaticAnalyser = (props: RemixUiStaticAnalyserProps) => {
...
+    props.event.trigger('staticAnaysisWarning', [
+      (slitherWarnings.length
+      + ssaWarnings.length
+      + hints.length) === 0 ? -1 : (slitherWarnings.length
+        + ssaWarnings.length
+        + hints.length)])
+  }
+  if(hideWarnings) {
```
Comment (LianaHus) [inline libs/remix-ui/static-analyser/src/lib/remix-ui-static-analyser.tsx:370]: space

File: libs/remix-ui/static-analyser/src/lib/remix-ui-static-analyser.tsx [hunk 12]
```diff
@@ -372,18 +350,106 @@ export const RemixUiStaticAnalyser = (props: RemixUiStaticAnalyserProps) => {
...
+  const hintsTotal = hints && solhintEnabled && state.data && state.source !== null ? hints.length : 0
+  props.event.trigger('staticAnaysisWarning', [(slitherTotal + ssaTotal + hintsTotal) === 0 ? -1 : (slitherTotal + ssaTotal + hintsTotal)])
+}, [solhintEnabled, basicEnabled, slitherEnabled, showSlither])
+
+  const handleSlitherEnabled = async () => {
+    const checkRemixd = await props.analysisModule.call('manager', 'isActive', 'remixd')
+    if (showSlither) {
+      setShowSlither(false)
+    }
+    if(!showSlither) {
```
Comment (LianaHus) [inline libs/remix-ui/static-analyser/src/lib/remix-ui-static-analyser.tsx:405]: space

File: libs/remix-ui/static-analyser/src/lib/remix-ui-static-analyser.tsx [hunk 13]
```diff
@@ -676,10 +734,10 @@ export const RemixUiStaticAnalyser = (props: RemixUiStaticAnalyserProps) => {
           <RemixUiCheckbox
             id="solhintstaticanalysis"
             inputType="checkbox"
-            title="Linter runs SolHint static analysis."
+            title="Solhint lints Solidity code for security and style guide validations"
```
Comment (LianaHus) [inline libs/remix-ui/static-analyser/src/lib/remix-ui-static-analyser.tsx:737]: please use a proper tooltip

File: libs/remix-ui/static-analyser/src/lib/remix-ui-static-analyser.tsx [hunk 14]
```diff
@@ -695,32 +753,36 @@ export const RemixUiStaticAnalyser = (props: RemixUiStaticAnalyserProps) => {
...
+                message, showWarnings, allWarnings, warningContainer, calculateWarningStateEntries, warningState, setHints, hints, setSlitherWarnings, setSsaWarnings, slitherEnabled, setStartAnalysis, solhintEnabled, basicEnabled)}
+              disabled={(state.data === null || !isSupportedVersion) || (!solhintEnabled && !basicEnabled) }
           />}
           {state && state.data !== null && state.source !== null && state.file.length > 0 ? (<div className="d-flex border-top flex-column">
-            {slitherWarnings.length > 0 || hints.length > 0 || Object.entries(warningState).length > 0 ?  (
-              <div className={`mt-4 p-2 d-flex ${slitherWarnings.length > 0 || hints.length > 0 || Object.entries(warningState).length > 0 ? 'border-top' : ''} flex-column`}>
+            {slitherWarnings.length > 0 || hints.length > 0 || ssaWarnings.length > 0 ?  (
+              <div className={`mt-4 p-2 d-flex ${slitherWarnings.length > 0 || hints.length > 0 || ssaWarnings.length > 0 ? 'border-top' : ''} flex-column`}>
                 <span>Last results for:</span>
                   <span
```
Comment (LianaHus) [inline libs/remix-ui/static-analyser/src/lib/remix-ui-static-analyser.tsx:779]: shift left pleas

File: libs/remix-ui/vertical-icons-panel/src/lib/components/Badge.tsx [hunk 15]
```diff
@@ -48,20 +48,34 @@ function Badge ({ badgeStatus }: BadgeProps) {
...
+                )
+              ) : (
+              badgeStatus.text
+            )}
+          </i>
+        </CustomTooltip>
+      ) : null}
     </>
-  )
+  );
```
Comment (LianaHus) [inline libs/remix-ui/vertical-icons-panel/src/lib/components/Badge.tsx:78]: do you need this ';' ?

**Review by LianaHus — CHANGES_REQUESTED** [review]
In case of 1 error 5 warnings the tooltip should say "1 error, 5 warnings"
Instead we see "6 warnings"

**Comment by LianaHus** [comment]
Badges on tool tabs titles should have the same style (including being centered, same width, and same text style)
see foto
Screenshot from 2023-06-29 11-11-15

**Comment by LianaHus** [comment]
The first tool tabs title should be aligned with alert element. see photo above

File: libs/remix-ui/static-analyser/src/lib/remix-ui-static-analyser.tsx (hunk 11 above)
Comment (joeizang) [reply libs/remix-ui/static-analyser/src/lib/remix-ui-static-analyser.tsx:370]: ok

File: libs/remix-ui/static-analyser/src/lib/remix-ui-static-analyser.tsx (hunk 10 above)
Comment (joeizang) [reply libs/remix-ui/static-analyser/src/lib/remix-ui-static-analyser.tsx:362]: done

File: apps/remix-ide-e2e/src/tests/staticAnalysis.test.ts [hunk 16]
```diff
@@ -34,7 +34,7 @@ module.exports = {
   'run analysis and filter results': function (browser: NightwatchBrowser) {
     browser
     .clickLaunchIcon('filePanel')
-    .click('*[data-id="treeViewLitreeViewItemcontracts"]')
+    .click('*[data-id="treeViewLitreeViewItemcontracts"]') //
```
[the remark this answers is not in the document]
Comment (joeizang) [reply apps/remix-ide-e2e/src/tests/staticAnalysis.test.ts:37]: done

File: apps/remix-ide/src/app/tabs/analysis-tab.js [hunk 17]
```diff
@@ -37,6 +37,16 @@ class AnalysisTab extends ViewPlugin {
         'offsettolinecolumnconverter').api
     }
     this.dispatch = null
+    /**
```
[the remark this answers is not in the document]
Comment (joeizang) [reply apps/remix-ide/src/app/tabs/analysis-tab.js:40]: done

**Comment by joeizang** [comment]
> In case of 1 error 5 warnings the tooltip should say "1 error, 5 warnings" Instead we see "6 warnings"

Nice idea. But can you create an issue so I work on it at a later date. This will delay the pr a bit

**Comment by LianaHus** [comment]
> > In case of 1 error 5 warnings the tooltip should say "1 error, 5 warnings" Instead we see "6 warnings"
> 
> Nice idea. But can you create an issue so I work on it at a later date. This will delay the pr a bit

:) it's not an idea for a new feature. the text we see here is wrong. this is a bug. I understand we need to merge and release but this is an issue we should fix before release.

**Comment by joeizang** [comment]
> https://deploy-preview-3838--remixproject.netlify.app

this is done. Based on our discussions at standup

**Review by Aniket-Engg — APPROVED** [review]

**Review by LianaHus — APPROVED** [review]

**Pull request merged**